src/libmain/loggers.cc

```diff
@@ -0,0 +1,51 @@
+#include "loggers.hh"
+#include "../nix/progress-bar.hh"
```

Comment:
This makes `libmain` depend on `nix`, which is a layering violation. It means you can't link against `libmain` anymore.

Reply:
Mh that's annoying :thinking: 

I moved `progress-bar` to `libmain` to fix that, but it also required moving `paths` to `libutil` to prevent `libmain` from depending on `libexpr`. Not sure if that's a great idea